src/xwayland/xwayland.c

```diff
@@ -287,10 +287,15 @@ wlc_xwayland_init(void)
          _exit(EXIT_FAILURE);
       }
 
+#ifdef HAVE_CLEARENV
       if (clearenv() != 0) {
          wlc_log(WLC_LOG_WARN, "Failed to clear environment");
          _exit(EXIT_FAILURE);
       }
+#else
+      extern char **environ;
+      environ[0] = NULL;
+#endif
```

Comment:
Can you turn these instances into own function, so there is no need for conditional #if's each time.